evalml/tests/component_tests/test_time_series_imputer.py

```diff
@@ -0,0 +1,478 @@
+        }
+    )
+    y_expected = pd.Series([0, 0, 1, 0, 1] * 4)
+    assert_frame_equal(X_expected, X_t, check_dtype=False)
+    assert_series_equal(y_expected, y_t, check_dtype=False)
+
+    imputer = TimeSeriesImputer()
+    transformed, _ = imputer.fit_transform(X, y)
+    assert_frame_equal(X_expected, transformed, check_dtype=False)
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_imputer_all_bool_return_original(data_type, make_data_type):
+    X = make_data_type(
+        data_type, pd.DataFrame([True, True, False, True, True], dtype=bool)
+    )
+    X_expected_arr = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    y = make_data_type(data_type, pd.Series([1, 0, 0, 1, 0]))
+
+    imputer = TimeSeriesImputer()
+    imputer.fit(X, y)
+    X_t, _ = imputer.transform(X)
+    assert_frame_equal(X_expected_arr, X_t)
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_imputer_bool_dtype_object(data_type, make_data_type):
+    X = pd.DataFrame([True, np.nan, False, np.nan, True] * 4)
+    y = pd.Series([1, 0, 0, 1, 0] * 4)
+    X_expected_arr = pd.DataFrame(
+        [True, True, False, False, True] * 4, dtype="category"
+    )
+    X = make_data_type(data_type, X)
+    y = make_data_type(data_type, y)
+    imputer = TimeSeriesImputer()
+    imputer.fit(X, y)
+    X_t, _ = imputer.transform(X)
+    assert_frame_equal(X_expected_arr, X_t)
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_imputer_multitype_with_one_bool(data_type, make_data_type):
+    X_multi = pd.DataFrame(
+        {
+            "bool with nan": pd.Series([True, np.nan, False, np.nan, False] * 4),
+            "bool no nan": pd.Series(
+                [False, False, False, False, True] * 4, dtype=bool
+            ),
+        }
+    )
+    y = pd.Series([1, 0, 0, 1, 0] * 4)
+    X_multi_expected_arr = pd.DataFrame(
+        {
+            "bool with nan": pd.Series(
+                [True, True, False, False, False] * 4, dtype="category"
+            ),
+            "bool no nan": pd.Series(
+                [False, False, False, False, True] * 4, dtype=bool
+            ),
+        }
+    )
+
+    X_multi = make_data_type(data_type, X_multi)
+    y = make_data_type(data_type, y)
+
+    imputer = TimeSeriesImputer()
+    imputer.fit(X_multi, y)
+    X_multi_t, _ = imputer.transform(X_multi)
+    assert_frame_equal(X_multi_expected_arr, X_multi_t)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        "int col",
+        "float col",
+        "categorical col",
+        "bool col",
+    ],
+)
+@pytest.mark.parametrize(
+    "logical_type", ["Integer", "Double", "Categorical", "NaturalLanguage", "Boolean"]
+)
+@pytest.mark.parametrize("has_nan", ["has_nan", "no_nans"])
+@pytest.mark.parametrize(
+    "numeric_impute_strategy", ["forwards_fill", "backwards_fill", "interpolate"]
+)
+def test_imputer_woodwork_custom_overrides_returned_by_components(
```

Comment:
I know we have this kind of test scattered throughout the components but honestly it's kind of a pain to reason through. I like the idea of parametrizing over the impute strategy but I would be happy if we took `imputer_test_data`, set the logical types to something where they don't match the default inference and then just check the logical types at the end are what they were at the beginning.

No need to parametrize over logical type and then try to catch all the possible incompatible conversions.